packages/aws-cdk-lib/aws-cloudfront/lib/vpc-origin.ts

```diff
@@ -0,0 +1,200 @@
+    });
+    return new VpcOriginEndpoint({ endpointArn, domainName: instance.instancePrivateDnsName });
+  }
+
+  /**
+   * A VPC origin endpoint from an Application Load Balancer.
+   * @param loadBalancer The Application Load Balancer as the CloudFront VPC origin endpoint.
+   */
+  public static fromApplicationLoadBalancer(loadBalancer: IApplicationLoadBalancer): VpcOriginEndpoint {
+    return new VpcOriginEndpoint({ endpointArn: loadBalancer.loadBalancerArn, domainName: loadBalancer.loadBalancerDnsName });
+  }
+
+  /**
+   * A VPC origin endpoint from an Network Load Balancer.
+   * @param loadBalancer The Network Load Balancer as the CloudFront VPC origin endpoint.
+   */
+  public static fromNetworkLoadBalancer(loadBalancer: INetworkLoadBalancer): VpcOriginEndpoint {
+    return new VpcOriginEndpoint({ endpointArn: loadBalancer.loadBalancerArn, domainName: loadBalancer.loadBalancerDnsName });
+  }
+
+  /**
+   * The ARN of the CloudFront VPC origin endpoint configuration.
+   */
+  readonly endpointArn: string;
+  /**
+   * The domain name of the CloudFront VPC origin endpoint configuration.
+   * @default - No domain name configured
+   */
+  readonly domainName?: string;
+
+  constructor(props: VpcOriginEndpointProps) {
+    this.endpointArn = props.endpointArn;
+    this.domainName = props.domainName;
+  }
+}
+
+/**
+ * A CloudFront VPC Origin configuration.
+ *
+ * @resource AWS::CloudFront::VpcOrigin
+ */
+export class VpcOrigin extends Resource implements IVpcOrigin {
+  /**
+   * Import an existing VPC origin from its ID.
+   */
+  public static fromVpcOriginId(scope: Construct, id: string, vpcOriginId: string): IVpcOrigin {
+    const vpcOriginArn = Stack.of(scope).formatArn({
+      service: 'cloudfront',
+      region: '',
+      resource: 'vpcorigin',
+      resourceName: vpcOriginId,
+    });
+
+    class Import extends Resource implements IVpcOrigin {
+      readonly vpcOriginArn = vpcOriginArn;
+      readonly vpcOriginId = vpcOriginId;
+      readonly domainName?: string = undefined;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The VPC origin ARN.
+   * @attribute
+   */
+  readonly vpcOriginArn: string;
+  /**
+   * The VPC origin ID.
+   * @attribute
+   */
+  readonly vpcOriginId: string;
+  /**
+   * The domain name of the CloudFront VPC origin endpoint configuration.
+   */
+  readonly domainName?: string;
+
+  constructor(scope: Construct, id: string, props: VpcOriginProps) {
+    super(scope, id);
+    // Enhanced CDK Analytics Telemetry
+    addConstructMetadata(this, props);
+
+    const resource = new CfnVpcOrigin(this, 'Resource', {
+      vpcOriginEndpointConfig: {
+        arn: props.endpoint.endpointArn,
+        httpPort: props.httpPort,
+        httpsPort: props.httpsPort,
+        name: props.vpcOriginName ?? Names.uniqueResourceName(this, {}),
+        originProtocolPolicy: props.protocolPolicy,
+        originSslProtocols: props.originSslProtocols ?? [OriginSslPolicy.TLS_V1_2],
```

Comment:
The default value of `originSslProtocols` is `['SSLv3', 'TLSv1']`.
It may be insecure, therefore I've overridden it by `['TLSv1.2']`.

Reply:
For details, see https://github.com/aws/aws-cdk/pull/33318#discussion_r1957514533